resources/design-document.md

```diff
@@ -0,0 +1,217 @@
+# Capstone Design Document
+
+---
+
+## 1. Problem Statement
+
+---
+
+People have to balance a lot of tasks when they cook. It's difficult for some to manually adjust recipes' ingredients to the amount they need versus the amount the recipe calls for. The goal of this app is to allow a user to build their own recipes and to provide them with the ability to easily adjust measurements according to the number of servings, switch between units of measurement, and create editable grocery lists based on the recipe's requirements.
+
+## 2. Top Questions to Resolve in Review
+
+---
+
+1. Would it be a good idea to cache the recipes that are called from the database?
+2. Does it make sense to store the ShoppingList items as a Map<Ingredient, Integer(quantity)> as opposed to 
+just having a Set<String(ingredientNames)>
+
+## 3. Use Cases
+
+---
+
+U1. As a customer, I want to create a new recipe
+
+U2. As a customer, I want to retrieve a specific recipe
+
+U3. As a customer, I want to be able to update a recipe with new values
+
+U4. As a customer, I want to create a list of ingredients that I can delete individual items from.
+
+U5. As a customer, I want variable units of measurement
+
+U6. As a customer, I want to be able to change amount of ingredients based on how many servings I want
+
+U7. As a customer, I want to sort recipes based on a certain ingredient
+
+U8. As a customer, I want to schedule meals ahead of time
+
+U9. As a customer, I want to pull recipes within a certain range of calories
```

Comment:
This seems like a nice feature - I see you have it in stretch, but I wonder if there's an easy way to have an MVP version of this (e.g. maybe a field that's just "num_calories") - different from a more advanced feature where you calculate the number of calories based on the ingredients (which seems a lot more involved).

Reply:
There is a 'calories' field in the Recipe class that I'm planning to do that with. It would rely on the user to come up with the value themselves, but it would be much simpler than storing a calorie value inside every ingredient and calculating from there.